src/starlite_saqlalchemy/scripts.py

```diff
@@ -41,6 +41,32 @@ async def _redis_ready() -> None:
             break
 
 
+def _get_uvicorn_config() -> uvicorn.Config:
+    reload = (
+        settings.server.RELOAD
+        if settings.server.RELOAD is not None
+        else settings.app.ENVIRONMENT == "local"
+    )
+    reload_dirs = settings.server.RELOAD_DIRS if reload else None
+
+    return uvicorn.Config(
+        app=settings.server.APP_LOC,
+        factory=settings.server.APP_LOC_IS_FACTORY,
+        host=settings.server.HOST,
+        loop="none",
+        port=settings.server.PORT,
+        reload=reload,
+        reload_dirs=reload_dirs,
+        timeout_keep_alive=settings.server.KEEPALIVE,
+    )
+
+
+async def _run_server(config: uvicorn.Config) -> None:
+    """Run an uvicorn server with the given config."""
+    server = uvicorn.Server(config)
+    await server.serve()
```

Comment:
Have you checked that this actually reloads?

In 7c93bd7d9edcebdbc03f0b9e448bb2d49dc5d92b I had to swap out `server.serve()` for `uvicorn.run()` as the reloading logic is in [`run()`](https://github.com/encode/uvicorn/blob/23b9f05a389a476aee151698cede171d915f6c53/uvicorn/main.py#L562-L569):

```python
    if config.should_reload:
        sock = config.bind_socket()
        ChangeReload(config, target=server.run, sockets=[sock]).run()
    elif config.workers > 1:
        sock = config.bind_socket()
        Multiprocess(config, target=server.run, sockets=[sock]).run()
    else:
        server.run()
```

This was to fix #178

Reply:
Tested, and no the reloading doesn't work under this scenario, but neither too does the application if reloading is off when using `uvicorn.run()` - so I've introduced a regression there.

I think this means we have to duplicate the above reloading/multiprocessing logic from `uvicorn.run()` in `_run_server()`...

Reply:
`uvicorn.run()` calls the sync method `Server.run()` to start the server. 

`Server.run()` uses `asyncio.run()` to run the coroutine `Server.serve()`.

`asyncio.run()` docs state:

> This function cannot be called when another asyncio event loop is running in the same thread.

We already have a loop running due to the service readiness checks.

Reply:
we should be doing the ready checks in the startup asgi lifespan handler, then we can leave uvicorn to manage the loop..